stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -271,25 +271,34 @@ end
 
 const PATH_cache_lock = Base.ReentrantLock()
 const PATH_cache = Set{String}()
-cached_PATH_string::Union{String,Nothing} = nothing
-function cached_PATH_changed()
-    global cached_PATH_string
-    @lock(PATH_cache_lock, cached_PATH_string) !== get(ENV, "PATH", nothing)
+PATH_cache_task::Union{Task,Nothing} = nothing # used for sync in tests
+next_cache_update::Float64 = 0.0
+function maybe_spawn_cache_PATH()
+    global PATH_cache_task, next_cache_update
+    @lock PATH_cache_lock begin
+        PATH_cache_task isa Task && !istaskdone(PATH_cache_task) && return
+        time() < next_cache_update && return
+    end
+    PATH_cache_task = Threads.@spawn REPLCompletions.cache_PATH()
+    Base.errormonitor(PATH_cache_task)
```

Comment:
```suggestion
        PATH_cache_task = Threads.@spawn REPLCompletions.cache_PATH()
        Base.errormonitor(PATH_cache_task)
    end
```
(to fix the data race here)